app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallViewModel.kt

```diff
@@ -80,20 +92,44 @@ class OngoingCallViewModel @Inject constructor(
     }
 
     fun hangUpCall() {
-        viewModelScope.launch {
+        uiScope.launch {
             endCall(conversationId)
             navigateBack()
         }
     }
 
+    fun setVideoPreview(view: View?) {
+        uiScope.launch {
+            setVideoPreview(conversationId, PlatformView(view))
+        }
+    }
+
+    private suspend fun pauseVideo() {
+        updateVideoState(conversationId, VideoState.PAUSED)
+        setVideoPreview(null)
+        callEstablishedState = callEstablishedState.copy(isCameraOn = false)
+    }
+
+    fun toggleVideo() {
+        uiScope.launch {
```

Comment:
what is the reason to launch all those coroutines ?

Reply:
`updateVideoState` is suspend